mysql/tests/test_metadata.py

```diff
@@ -30,3 +47,484 @@ def test_collect_mysql_settings(aggregator, dbm_instance, dd_run_check):
+                    }
+                ],
+                "indexes": [
+                    {
+                        "name": "FK_CityId",
+                        "collation": "A",
+                        "cardinality": "0",
+                        "index_type": "BTREE",
+                        "seq_in_index": "1",
+                        "columns": "city_id",
+                        "sub_parts": "None",
+                        "packed": "None",
+                        "nullables": "true",
+                        "non_uniques": "1",
+                    }
+                ],
+            },
+        ],
+    }
+    exp_datadog_test_schemas_second = {
+        "name": "datadog_test_schemas_second",
+        "default_character_set_name": "latin1",
+        "default_collation_name": "latin1_swedish_ci",
+        "tables": [
+            {
+                "name": "ϑings",
+                "columns": [
+                    {"name": "id", "data_type": "int", "default": "0", "nullable": True, "ordinal_position": "1"},
+                    {
+                        "name": "name",
+                        "data_type": "varchar",
+                        "default": "None",
+                        "nullable": True,
+                        "ordinal_position": "2",
+                    },
+                ],
+                "indexes": [
+                    {
+                        "name": "thingsindex",
+                        "collation": "A",
+                        "cardinality": "2",
+                        "index_type": "BTREE",
+                        "seq_in_index": "1",
+                        "columns": "name",
+                        "sub_parts": "None",
+                        "packed": "None",
+                        "nullables": "true",
+                        "non_uniques": "0",
+                    }
+                ],
+            },
+            {
+                "name": "ts",
+                "columns": [
+                    {"name": "id", "data_type": "int", "default": "None", "nullable": True, "ordinal_position": "1"},
+                    {
+                        "name": "purchased",
+                        "data_type": "date",
+                        "default": "None",
+                        "nullable": True,
+                        "ordinal_position": "2",
+                    },
+                ],
+                "partitions": [
+                    {
+                        "name": "p0",
+                        "subpartition_names": "p0sp0,p0sp1",
+                        "partition_ordinal_position": "1",
+                        "subpartition_ordinal_positions": "1,2",
+                        "partition_method": "RANGE",
+                        "subpartition_methods": "HASH,HASH",
+                        "partition_expression": " YEAR(purchased)",
+                        "subpartition_expressions": " TO_DAYS(purchased), TO_DAYS(purchased)",
```

Comment:
I concatenated sub-partitions but not partitions.
I don't have a strong opinion about that. I felt that otherwise there would be too many concatenations and it wont be readable. On a flip side having 20 partitions json fields would make us sent too much data. Not sure whats the best.